src/webgpu/shader/validation/parse/if.spec.ts

```diff
@@ -0,0 +1,63 @@
+export const description = `Validation parser tests for 'if' statements`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kTests = {
+  true: { wgsl: `if true {}`, pass: true },
+  paren_true: { wgsl: `if (true) {}`, pass: true },
+  expr: { wgsl: `if expr {}`, pass: true },
+  paren_expr: { wgsl: `if (expr) {}`, pass: true },
+
+  true_else: { wgsl: `if true {} else {}`, pass: true },
+  paren_true_else: { wgsl: `if (true) {} else {}`, pass: true },
+  expr_else: { wgsl: `if expr {} else {}`, pass: true },
+  paren_expr_else: { wgsl: `if (expr) {} else {}`, pass: true },
+
+  true_else_if_true: { wgsl: `if true {} else if true {}`, pass: true },
+  paren_true_else_if_paren_true: { wgsl: `if (true) {} else if (true) {}`, pass: true },
+  true_else_if_paren_true: { wgsl: `if true {} else if (true) {}`, pass: true },
+  paren_true_else_if_true: { wgsl: `if (true) {} else if true {}`, pass: true },
+
+  expr_else_if_expr: { wgsl: `if expr {} else if expr {}`, pass: true },
+  paren_expr_else_if_paren_expr: { wgsl: `if (expr) {} else if (expr) {}`, pass: true },
+  expr_else_if_paren_expr: { wgsl: `if expr {} else if (expr) {}`, pass: true },
+  paren_expr_else_if_expr: { wgsl: `if (expr) {} else if expr {}`, pass: true },
+
+  if: { wgsl: `if`, pass: false },
+  block: { wgsl: `if{}`, pass: false },
+  semicolon: { wgsl: `if;`, pass: false },
+  true_lbrace: { wgsl: `if true {`, pass: false },
+  true_rbrace: { wgsl: `if true }`, pass: false },
+
+  lparen_true: { wgsl: `if (true {}`, pass: false },
+  rparen_true: { wgsl: `if )true {}`, pass: false },
+  true_lparen: { wgsl: `if true( {}`, pass: false },
+  true_rparen: { wgsl: `if true) {}`, pass: false },
+
+  true_else_if_no_block: { wgsl: `if true {} else if `, pass: false },
+  true_else_if: { wgsl: `if true {} else if {}`, pass: false },
+  true_else_if_semicolon: { wgsl: `if true {} else if ;`, pass: false },
+  true_else_if_true_lbrace: { wgsl: `if true {} else if true {`, pass: false },
+  true_else_if_true_rbrace: { wgsl: `if true {} else if true }`, pass: false },
+
+  true_else_if_lparen_true: { wgsl: `if true {} else if (true {}`, pass: false },
+  true_else_if_rparen_true: { wgsl: `if true {} else if )true {}`, pass: false },
+  true_else_if_true_lparen: { wgsl: `if true {} else if true( {}`, pass: false },
+  true_else_if_true_rparen: { wgsl: `if true {} else if true) {}`, pass: false },
```

Comment:
Please add a cases:
*  ` else { }` 
*  ` else if true { }`
and because we can, please add cases from other languages:
*  `if (true) { } elif (true) {}`
*  `if (true) { } elsif (true) {}`
*  `elif (true) {}`
*  `elsif (true) {}`

Reply:
Done